core/editor/editor-loader.php

```diff
@@ -0,0 +1,112 @@
+<?php
+namespace Elementor\Core\Editor;
+
+use Elementor\Core\Utils\Collection;
+use Elementor\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Editor_Loader {
+	const VERSION_1 = 1;
+	const VERSION_2 = 2;
+
+	const AVAILABLE_VERSIONS = [
+		self::VERSION_1,
+		self::VERSION_2,
+	];
+
+	/**
+	 * @var number
+	 */
+	private $post_id;
+
+	/**
+	 * @var number
+	 */
+	private $version;
+
+	/**
+	 * @param int|string $post_id
+	 * @param int|string $version
+	 */
+	public function __construct( $post_id, $version = self::VERSION_1 ) {
+		$this->post_id = (int) $post_id;
+		$this->version = in_array( (int) $version, self::AVAILABLE_VERSIONS, true )
```

Comment:
maybe instead of passing a version, we can pass an interface of some Provider, one of each version.
The provider will (probably?) be responsible only for retrieving data (e.g. styles/scripts arrays, templates, config, etc.) and this editor loader will consume the data and contain the actual logic of register/enqueue/load templates. 
WDYT?

Reply:
Done